keras_cv/models/segmentation/fcn_test.py

```diff
@@ -0,0 +1,205 @@
+        print(type(backbone))
+        model = FCN(classes=11, backbone=backbone, input_shape=(256, 256, 3))
+
+        input_image = tf.random.uniform(shape=[2, 256, 256, 3])
+        output = model(input_image)
+
+        self.assertEquals(output.shape, [2, 256, 256, 11])
+
+    def test_mixed_precision(self):
+        tf.keras.mixed_precision.set_global_policy("mixed_float16")
+        model = FCN(
+            classes=11,
+            backbone="vgg16",
+            model_architecture="fcn8s",
+            input_shape=(256, 256, 3),
+        )
+        input_image = tf.random.uniform(shape=[2, 256, 256, 3])
+        output = model(input_image, training=True)
+
+        self.assertEquals(output.dtype, tf.float32)
+
+    def test_invalid_backbone_model(self):
+        with self.assertRaisesRegex(
+            ValueError,
+            r"Chosen `backbone` argument is not a valid allowed backbone. Possible options are ['vgg16', 'vgg19']"
+        ):
+            FCN(
+                classes=11,
+                backbone="resnet",
+                model_architecture="fcn8s",
+                input_shape=(256, 256, 3),
+            )
+        with self.assertRaisesRegex(
+            ValueError,
+            r"Invalid argument for parameter `backbone`. Accepted values are ['vgg16', 'vgg19'] or a `tf.keras.models.Model` instance with only `tf.keras.layers.Conv2D`, 'tf.keras.layers.MaxPooling2D' or `tf.keras.layers.Dense` layers",
+        ):
+            FCN(
+                classes=11,
+                backbone=tf.Module(),
+                model_architecture='fcn8s'
+            )
+        with self.assertRaisesRegex(
+            ValueError,
+            r"Entered `backbone` argument has custom layers. Include a `tf.keras.models.Model` with `keras.layers.Conv2D` or `keras.layers.MaxPooling2D` layers only.",
+        ):
+            input_tensor = tf.keras.Input(shape=(256, 256, 3))
+            output_tensor = tf.keras.layers.AveragePooling2D(input_tensor)
+            backbone_model = tf.keras.Model(inputs=input_tensor, outputs=output_tensor)
+            FCN(
+                classes=11,
+                backbone=backbone_model,
+            )
+
+    def test_invalid_model_architecture(self):
+        with self.assertRaisesRegex(
+            ValueError,
+            r"Invalid argument for parameter `model_architecture`. Accepted values are ['fcn8s', 'fcn16s', 'fcn32s']"
+        ):
+            FCN(
+                classes=11,
+                backbone="vgg16",
+                model_architecture="fcn10s",
+                input_shape=(256, 256, 3),
+            )
+
+    @pytest.mark.skipif(
+        "INTEGRATION" not in os.environ or os.environ["INTEGRATION"] != "true",
+        reason="Takes a long time to run, only runs when INTEGRATION "
+        "environment variable is set.  To run the test please run: \n"
+        "`INTEGRATION=true pytest keras_cv/",
+    )
+    def test_model_train(self):
```

Comment:
I have not, actually

Reply:
Since integration tests only run rarely, we won't know whether this breaks the CI unless we verify that it runs upfront (either by including the integration test or by running at least locally). It won't show a failed run before merging

Reply:
We can also just note to run the integration test, no need to do it in a separate script?